compiler/noirc_evaluator/src/ssa/opt/mem2reg/block.rs

```diff
@@ -164,6 +164,9 @@ impl Block {
                 // known references if this reference is ever stored to.
             }
         }
+        // else {
+        //     self.references.insert(result, ReferenceValue::Known(address));
+        // }
```

Comment:
I think we don't handle these `Expression::Dereference` cases well in mem2reg in general.